spinetoolbox/widgets/conda_envs.py

```diff
@@ -0,0 +1,103 @@
+######################################################################################################################
+
+"""
+Conda environment widget.
+
+:author: P. Savolainen (VTT)
+:date:   19.3.2021
+"""
+from PySide2.QtWidgets import QWidget
+from PySide2.QtCore import Slot, Qt
+from PySide2.QtGui import QStandardItemModel, QStandardItem
+from spine_engine.execution_managers.conda_kernel_spec_manager import CondaKernelSpecManager
+# import environment_kernels
+# from environment_kernels.envs_conda import get_conda_env_data
+
+
+class CondaEnv(QWidget):
+    """Class for a Conda environment editor."""
+
+    def __init__(self, parent):
+        """
+
+        Args:
+            parent (QWidget): Parent widget (Settings widget)
+        """
+        from ..ui import conda_env_window  # pylint: disable=import-outside-toplevel
+
+        super().__init__(parent=parent)  # Inherits stylesheet from SettingsWindow
+        self.setWindowFlags(Qt.Window)
+        self._parent = parent  # QSettingsWidget
+        self.ui = conda_env_window.Ui_Form()
+        self.ui.setupUi(self)
+        self.setAttribute(Qt.WA_DeleteOnClose)
+        self.connect_signals()
+        self.ksm = CondaKernelSpecManager()
+        self.model = QStandardItemModel()
+        self.ui.listView.setModel(self.model)
+        all_specs = self.ksm._all_specs()
+        self.populate_model(all_specs.keys())
+        # self.ksm.conda_env_dirs.append("~/.julia/conda/3/envs/")
+
+    def connect_signals(self):
+        """Connect signals."""
+        self.ui.buttonBox.accepted.connect(self.close)
+        self.ui.buttonBox.rejected.connect(self.close)
+        self.ui.toolButton_refresh.clicked.connect(self.refresh)
+
+    def populate_model(self, items):
+        self.model.clear()
+        for i in items:
+            item = QStandardItem(i)
+            self.model.appendRow(item)
+
+    @Slot(bool)
+    def refresh(self, _checked=False):
+        self.model.clear()
+        kernel_specs = self.ksm.find_kernel_specs()
+        print("All specs\n")
+        for k, v in kernel_specs.items():
+            print(f"{k}: {v}")
+
+        print("\nAll envs\n")
+        all_envs = self.ksm._all_envs()
+        for k, v in all_envs.items():
+            print(f"{k}: {v}")
+
+        print("\nAll specs\n")
+        all_specs = self.ksm._all_specs()
+        for k, v in all_specs.items():
+            print(f"{k}: {v}")
+        self.populate_model(all_specs.keys())
+
+        # kernel_specs_for_envs = self.ksm.find_kernel_specs_for_envs()
+        # all_kernel_specs_for_envs = self.ksm.get_all_kernel_specs_for_envs()
+        # print("All kernel specs\n")
+        # for k, v in all_kernel_specs_for_envs.items():
+        #     print(f"{k}: {v.to_dict()}")
+        #     print(f"\nenv:{v.env}\n")
+        # conda_data = get_conda_env_data(self.ksm)
+        # for a, b in conda_data.items():
+        #     print(f"{a}: {b}")
+
+    def keyPressEvent(self, e):
```

Comment:
I would rather see this as a `QAction`, i.e. add a 'Close' action with correct keyboard shortcut to the `.ui` file in Designer, and connect its `triggered` signal to `close` slot in `__init__.py`.

Reply:
conda_envs.py file was not used so I removed it.